src/modules/scraper/adapter/messaging/TrackFillEventConsumer.ts

```diff
@@ -0,0 +1,94 @@
+import { OnQueueFailed, Process, Processor } from "@nestjs/bull";
+import { Logger } from "@nestjs/common";
+import { Job } from "bull";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+import { utils } from "ethers";
+import { TrackFillEventQueueMessage, ScraperQueue } from ".";
+import { Deposit } from "../../model/deposit.entity";
+import { TrackService } from "../amplitude/track-service";
+import { makeWeiPctValuesFormatter } from "../amplitude/utils";
+import { ChainIdToName } from "../../../web3/model/ChainId";
+import BigNumber from "bignumber.js";
+
+@Processor(ScraperQueue.TrackFillEvent)
+export class TrackFillEventConsumer {
+  private logger = new Logger(TrackFillEventConsumer.name);
+
+  constructor(
+    private trackService: TrackService,
+    @InjectRepository(Deposit) private depositRepository: Repository<Deposit>,
+  ) {}
+
+  @Process()
+  private async process(job: Job<TrackFillEventQueueMessage>) {
+    const { depositId } = job.data;
+    const deposit = await this.depositRepository.findOne({ where: { id: depositId }, relations: ["token", "price"] });
+
+    if (!deposit) return;
+    if (deposit.status !== "filled") return;
+
+    if (!deposit.token || !deposit.price) {
+      throw new Error("Can not track fill event without token or price");
+    }
+
+    const amountFormatted = utils.formatUnits(deposit.amount, deposit.token.decimals);
+    const weiPctValuesFormatter = makeWeiPctValuesFormatter(amountFormatted, deposit.price.usd);
+    const formattedLpFeeValues = weiPctValuesFormatter(deposit.realizedLpFeePct);
+    const formattedRelayFeeValues = weiPctValuesFormatter(deposit.depositRelayerFeePct);
+    const formattedBridgeFeeValues = weiPctValuesFormatter(deposit.bridgeFeePct);
+
+    this.trackService.trackDepositFilledEvent(deposit.depositorAddr, {
+      capitalFeePct: "-",
+      capitalFeeTotal: "-",
+      capitalFeeTotalUsd: "-",
```

Comment:
Is it possible to derive these values at the time a fill event is indexed?

Reply:
@nicholaspai

Reply:
You cannot exactly. You can get close if you know the gas price at the time the event was indexed. The [capital fee is a math formula](https://github.com/across-protocol/sdk-v2/blob/master/src/relayFeeCalculator/relayFeeCalculator.ts#L123) that isn't time dependent but it depends on a `capitalCostsConfig` that is stored on the API that quotes the fee.  The gas fee depends on the [gas price and gas token price](https://github.com/across-protocol/sdk-v2/blob/master/src/relayFeeCalculator/relayFeeCalculator.ts#L103) at the time of indexing

Reply:
We now derive it by subtracting the tx fee from the relayer fee. See https://github.com/across-protocol/scraper-api/pull/186/commits/50eb98713e8ff05b07b532b00ce1971e55220b97